src/System.Private.CoreLib/shared/System/Buffer.cs

```diff
@@ -158,7 +160,7 @@ internal static unsafe void Memmove(byte* dest, byte* src, nuint len)
             // Copy bytes which are multiples of 16 and leave the remainder for MCPY01 to handle.
             Debug.Assert(len > 16 && len <= 64);
 #if HAS_CUSTOM_BLOCKS
-            *(Block16*)dest = *(Block16*)src;                   // [0,16]
+            Unsafe.AsRef<Vector128<byte>>(dest) = Unsafe.AsRef<Vector128<byte>>(src); // [0,16]
```

Comment:
We already have a pointer, why not explicitly `Load`. That would more easily allow use of the non-temporal loads, if desirable (and make it easier to test them).

Reply:
Don't know if non-temporal is desirable. 

Also would mean every copy would need to be doubled and guarded with an `.IsSupported` rather than just working?

Reply:
> Also would mean every copy would need to be doubled and guarded with an .IsSupported rather than just working?

I think providing a `Load` helper method could be used to handle that, and abstract away the `x86` vs `Arm` vs fallback cases.